webui.py

```diff
@@ -100,10 +100,7 @@ def sigint_handler(sig, frame):
     
     if cmd_opts.api:
         from modules.api.api import Api
-        api = Api(txt2img=modules.txt2img.txt2img,
-          img2img=modules.img2img.img2img,
-          run_extras=modules.extras.run_extras,
-          run_pnginfo=modules.extras.run_pnginfo)
+        api = Api()
 
         api.launch(server_name="0.0.0.0" if cmd_opts.listen else "127.0.0.1",

```

Comment:
Since gradio itself is already using FastAPI, wouldn't it be much easier; and possibly decrease future headaches to combine the api with gradio itself? Looks like a neater way to do this rather than running two independent fastapi instances within the same program.  The gradio docs show how you can do this nicely https://www.gradio.app/docs/#mount_gradio_app